src/arch/xtensa/smp/include/arch/idc.h

```diff
@@ -280,6 +280,8 @@ static inline void idc_do_cmd(void *data)
 
 	idc_cmd(&idc->received_msg);
 
+	schedule_task_complete(&idc->idc_task);
```

Comment:
I don't think we want to be changing scheduler task state outside of scheduler/task subsystems.